storage_policies/stack/__main__.py

```diff
@@ -186,3 +186,77 @@ def add_bucket_permissions(name: str,
                            release_access_group,
                            release_bucket,
                            'roles/storage.objectViewer')
+
+# Cloud Run is used for the server component of the analysis-runner.
+cloudrun = gcp.projects.Service('cloudrun-service',
+                                service='run.googleapis.com',
+                                disable_on_destroy=False)
+
+analysis_runner_service_account = gcp.serviceaccount.Account(
+    'analysis-runner-service-account',
+    account_id='analysis-runner-server',
+    display_name='analysis-runner service account',
+    opts=pulumi.resource.ResourceOptions(depends_on=[cloudidentity]))
+
+secretmanager = gcp.projects.Service('secretmanager-service',
+                                     service='secretmanager.googleapis.com',
+                                     disable_on_destroy=False)
+
+# Add a secret that will hold the Hail token.
+hail_token_secret = gcp.secretmanager.Secret(
+    'hail-token-secret',
+    replication=gcp.secretmanager.SecretReplicationArgs(
+        user_managed=gcp.secretmanager.SecretReplicationUserManagedArgs(
+            replicas=[
+                gcp.secretmanager.SecretReplicationUserManagedReplicaArgs(
+                    location=REGION
+                ),
+            ],
+        ),
+    ),
+    secret_id='hail-token',
+    opts=pulumi.resource.ResourceOptions(depends_on=[secretmanager]))
+
+# The analysis-runner server needs to read the Hail token secret.
+gcp.secretmanager.SecretIamMember(
+    'analysis-runner-service-account-secret-reader',
+    secret_id=hail_token_secret.id,
+    role='roles/secretmanager.secretAccessor',
+    member=pulumi.Output.concat('serviceAccount:', analysis_runner_service_account.email))
+
+project_number = gcp.organizations.get_project().number
+
+# Allow the Cloud Run Service Agent to pull the image.
+repo_member = gcp.artifactregistry.RepositoryIamMember(
+    'artifact-registry-reader',
+    project='analysis-runner',
```

Comment:
What is "project" here - is it a GCP project id? If yes, wondering why should it be different from the global project?

Reply:
Yes, this is a GCP project ID. The global project will be something like "tob-wgs", referring to the dataset. However, the image is currently only stored once, in the "analysis-runner" project's Artifact Registry repository. This lets the Cloud Run service account from "tob-wgs" access the image from the "analysis-runner" project.

I've updated the code comment to make this more clear.

Reply:
Right, that makes sense! Thanks :)